pkg/detectors/easyinsight/easyinsight.go

```diff
@@ -38,49 +38,48 @@ func (s Scanner) Keywords() []string {
 func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
 	dataStr := string(data)
 
-	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+	var keyMatches, idMatches = make(map[string]struct{}), make(map[string]struct{})
 
-	idmatches := idPat.FindAllStringSubmatch(dataStr, -1)
+	// get unique key and id matches
+	for _, matches := range keyPat.FindAllStringSubmatch(dataStr, -1) {
+		keyMatches[matches[1]] = struct{}{}
+	}
 
-	for _, match := range matches {
-		if len(match) != 2 {
-			continue
-		}
-		resMatch := strings.TrimSpace(match[1])
-		for _, idmatch := range idmatches {
-			if len(idmatch) != 2 {
+	for _, matches := range idPat.FindAllStringSubmatch(dataStr, -1) {
+		idMatches[matches[1]] = struct{}{}
+	}
+
+	for keyMatch := range keyMatches {
+		for idMatch := range idMatches {
+			//as key and id regex are same, the strings captured by both regex will be same.
+			//avoid processing when key is same as id. This will allow detector to process only different combinations
+			if keyMatch == idMatch {
 				continue
 			}
-			resIdMatch := strings.TrimSpace(idmatch[1])
 
 			s1 := detectors.Result{
 				DetectorType: detectorspb.DetectorType_EasyInsight,
-				Raw:          []byte(resMatch),
-				RawV2:        []byte(resMatch + resIdMatch),
+				Raw:          []byte(keyMatch),
+				RawV2:        []byte(keyMatch + idMatch),
 			}
 
 			if verify {
-				data := fmt.Sprintf("%s:%s", resIdMatch, resMatch)
-				sEnc := b64.StdEncoding.EncodeToString([]byte(data))
-				req, err := http.NewRequestWithContext(ctx, "GET", "https://www.easy-insight.com/app/api/users.json", nil)
-				if err != nil {
-					continue
-				}
-				req.Header.Add("Content-Type", "application/json")
-				req.Header.Add("Accept", "application/json")
-				req.Header.Add("Authorization", fmt.Sprintf("Basic %s", sEnc))
-				res, err := client.Do(req)
-				if err == nil {
-					defer res.Body.Close()
-					if res.StatusCode >= 200 && res.StatusCode < 300 {
-						s1.Verified = true
-					}
+				auth := fmt.Sprintf("%s:%s", idMatch, keyMatch)
+				sEnc := b64.StdEncoding.EncodeToString([]byte(auth))
```

Comment:
Can be simplified with https://pkg.go.dev/net/http#Request.SetBasicAuth

Reply:
`id` as username and `key` as password in Basic Auth?

Reply:
Yeah, rather than manually base64 encoding.